### PR TITLE
Changes for new Droid and backend checks

### PR DESCRIPTION
**Pull request by MancunianSam**
* Move the lambda to use the new Droid library
* Remove the SQS queues

### CODE REVIEW

File: project/Dependencies.scala [hunk 1]
```diff
@@ -2,26 +2,26 @@ import sbt._
...
   lazy val logback = "ch.qos.logback" % "logback-classic" % "1.4.5"
   lazy val logstashLogbackEncoder = "net.logstash.logback" % "logstash-logback-encoder" % "7.2"
   lazy val mockito = "org.mockito" %% "mockito-scala" % "1.17.12"
-  lazy val elasticMq = "org.elasticmq" %% "elasticmq-server" % elasticMqVersion
-  lazy val elasticMqSqs = "org.elasticmq" %% "elasticmq-rest-sqs" % elasticMqVersion
   lazy val wiremock = "com.github.tomakehurst" % "wiremock" % "2.27.2"
+  lazy val apacheCommons = "org.apache.commons" % "commons-lang3" % "3.12.0"
+  lazy val droidApi = "uk.gov.nationalarchives" % "droid-api" % "6.6.0-rc2"
+  // This is an older version of this dependency but the newer version won't work with Droid without some major changes.
+  // I'll configure Scala Steward to ignore it.
```
Comment (TomJKing) [inline project/Dependencies.scala:25]: Maybe change this second comment to: `Scala Steward configured to ignore it`

File: project/Dependencies.scala [hunk 2]
```diff
@@ -2,26 +2,26 @@ import sbt._
...
   lazy val scalaLogging = "com.typesafe.scala-logging" %% "scala-logging" % "3.9.5"
   lazy val logback = "ch.qos.logback" % "logback-classic" % "1.4.5"
   lazy val logstashLogbackEncoder = "net.logstash.logback" % "logstash-logback-encoder" % "7.2"
   lazy val mockito = "org.mockito" %% "mockito-scala" % "1.17.12"
-  lazy val elasticMq = "org.elasticmq" %% "elasticmq-server" % elasticMqVersion
-  lazy val elasticMqSqs = "org.elasticmq" %% "elasticmq-rest-sqs" % elasticMqVersion
   lazy val wiremock = "com.github.tomakehurst" % "wiremock" % "2.27.2"
+  lazy val apacheCommons = "org.apache.commons" % "commons-lang3" % "3.12.0"
+  lazy val droidApi = "uk.gov.nationalarchives" % "droid-api" % "6.6.0-rc2"
+  // This is an older version of this dependency but the newer version won't work with Droid without some major changes.
```
Comment (TomJKing) [inline project/Dependencies.scala:24]: Is there anything on the `DROID` backlog about updating this?

File: src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker [hunk 3]
```diff
@@ -0,0 +1 @@
+mock-maker-inline
```
Comment (TomJKing) [inline src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker:1]: Missing line break at end of file.

Comment (TomJKing) [inline src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker:1]: Missing line break at end of file.

**Review by TomJKing — COMMENTED** [review]
Added some minor comments.

I tried to run locally getting an error when trying to download the DROID snapshot version?
```
[error] sbt.librarymanagement.ResolveException: Error downloading uk.gov.nationalarchives:droid-api:6.6.0-SNAPSHOT
[error]   Not found

```

Some other general comments:

- should update the README to take account of the new architecture
- there are a couple of `Jenkins` files which I think can be deleted from the repo now

File: project/Dependencies.scala (hunk 2 above)
Comment (MancunianSam) [reply project/Dependencies.scala:24]: I'm not sure, from the brief look I had, it's a large job so it won't be any time soon. They're getting a contractor in to look at Droid soon though so maybe then.

**Comment by MancunianSam** [comment]
I'm not sure why you're getting not found on the snapshot version. Can you try clearing out the target directory and trying again, it sounds like something weird is cached somewhere.

I've removed the Jenkinsfiles. I've also removed the docker image which writes Droid to EFS, the job which deploys the docker image and the job that runs it. 

I've changed the README as well.

File: project/Dependencies.scala (hunk 2 above)
Comment (TomJKing) [reply project/Dependencies.scala:24]: Fair enough, was just wondering if there was a task we could reference about this.

File: src/main/scala/uk/gov/nationalarchives/fileformat/LambdaRunner.scala [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+import com.amazonaws.services.lambda.runtime.events.SQSEvent
+import com.amazonaws.services.lambda.runtime.events.SQSEvent.SQSMessage
+import org.apache.commons.io.output.ByteArrayOutputStream
+
+import java.io.ByteArrayInputStream
+import scala.jdk.CollectionConverters._
+
+object LambdaRunner extends App {
+    val event = new SQSEvent()
```
Comment (TomJKing) [inline src/main/scala/uk/gov/nationalarchives/fileformat/LambdaRunner.scala:11]: Is this `val event` used anywhere?

File: src/main/scala/uk/gov/nationalarchives/fileformat/LambdaRunner.scala [hunk 5]
```diff
@@ -0,0 +1,17 @@
+package uk.gov.nationalarchives.fileformat
+
+import com.amazonaws.services.lambda.runtime.events.SQSEvent
+import com.amazonaws.services.lambda.runtime.events.SQSEvent.SQSMessage
```
Comment (TomJKing) [inline src/main/scala/uk/gov/nationalarchives/fileformat/LambdaRunner.scala:4]: Not sure this import is needed? `import com.amazonaws.services.lambda.runtime.events.SQSEvent.SQSMessage`

File: src/test/scala/uk/gov/nationalarchives/fileformat/FileSpec.scala [hunk 6]
```diff
@@ -0,0 +1,49 @@
+package uk.gov.nationalarchives.fileformat
+
+import com.github.tomakehurst.wiremock.WireMockServer
+import org.scalatest.{BeforeAndAfterAll, BeforeAndAfterEach, Suite}
+import com.github.tomakehurst.wiremock.WireMockServer
```
Comment (TomJKing) [inline src/test/scala/uk/gov/nationalarchives/fileformat/FileSpec.scala:5]: Duplicate import with line 3: `import com.github.tomakehurst.wiremock.WireMockServer`

File: src/test/scala/uk/gov/nationalarchives/fileformat/FileSpec.scala [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+      .willReturn(okXml(fromResource("containers/container_signatures.xml").mkString)))
+    tnaCdn
+  }
+
+  override def beforeAll(): Unit = {
+    tnaCdn.start()
+    super.beforeAll()
+  }
+
+  // Create the files and then try to ls them in the test script. This mimics reading from the existing file on EFS
```
Comment (TomJKing) [inline src/test/scala/uk/gov/nationalarchives/fileformat/FileSpec.scala:27]: I thought we no longer need EFS? Maybe this comment needs updating?

File: src/test/scala/uk/gov/nationalarchives/fileformat/LambdaTest.scala [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+import com.github.tomakehurst.wiremock.client.WireMock.{aResponse, anyUrl, get, okXml, urlEqualTo}
+import graphql.codegen.types.FFIDMetadataInput
+import io.circe.{DecodingFailure, Printer}
+import io.circe.generic.auto._
+import io.circe.parser.decode
+import io.circe.syntax._
+import org.scalatest.{BeforeAndAfterAll, BeforeAndAfterEach}
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers.{equal, _}
+import org.scalatest.prop.{TableDrivenPropertyChecks, TableFor1, TableFor2}
```
Comment (TomJKing) [inline src/test/scala/uk/gov/nationalarchives/fileformat/LambdaTest.scala:13]: Unused import: `import org.scalatest.prop.{..., TableFor1, ...}`

File: README.md [hunk 9]
```diff
@@ -4,30 +4,24 @@ This is the repository for the backend file format check for the [Transfer Digit
...
 
-The [build GitHub actions job] will run when a pull request is merged to master. 
-The job will check to see if there are any changes to the Dockerfile. 
-If there are, it will build a new version of the docker image and push it to ECS.
-
-To run the new image, run the [Run ECS Task] workflow with the latest docker image version and the environment you are deploying to.
+Droid is contained within the `"uk.gov.nationalarchives" % "droid-api"` dependency. 
+The binary and container signature files are downloaded when the lambda is first started. 
+The versions of the signature files are stored in the `DROID_SIGNATURE_VERSION` and `CONTAINER_SIGNATURE_VERSION` environment variables. 
+To update to a new version of these signature files, we need to update the values in the environment variables.
```
Comment (TomJKing) [inline README.md:23]: Is this a manual process to find out if there is a new version? Probably should note where to find the latest versions.

**Review by TomJKing — COMMENTED** [review]
Some additional comments. Clearing out the cache sorted out downloading the `DROID` snapshot version.

File: src/main/scala/uk/gov/nationalarchives/fileformat/LambdaRunner.scala (hunk 4 above)
Comment (MancunianSam) [reply src/main/scala/uk/gov/nationalarchives/fileformat/LambdaRunner.scala:11]: No, it's not, I've removed it.

File: src/main/scala/uk/gov/nationalarchives/fileformat/LambdaRunner.scala (hunk 5 above)
Comment (MancunianSam) [reply src/main/scala/uk/gov/nationalarchives/fileformat/LambdaRunner.scala:4]: It's not needed and the entire library isn't needed. I've removed this and the AWS library they're imported from.

File: README.md (hunk 9 above)
Comment (MancunianSam) [reply README.md:23]: It is a manual process at the moment. I'm hoping there will be some kind of message we can subscribe to to automatically update them but there's nothing yet. I've put this in the README along with where to find the signature files.

**Review by TomJKing — APPROVED** [review]
Looks good. Thanks

**Pull request merged**